### PR TITLE
Create list of candidate 

### PR DESCRIPTION
**Pull request by ValeryiaMIRON**
- Add AG Grid to our Project
- Add AG-theme
- Enable Sorting , Filtering, Sizing
- Add Column Dragging
- Add Column Menu and Tool Panel(Side Bar)
- Add Column Sizing
- Add DOM Layout print (no scroll bars are used and the grid renders all rows and columns)
- Add checkboxSelection
- Add rowSelection multiple
- Add menu for copy, CSV export and Excel export
- Add pagination
- Add routing

Снимок экрана 2021-10-31 в 3 09 25 PM

Снимок экрана 2021-10-31 в 3 11 07 PM

Снимок экрана 2021-10-28 в 8 43 59 PM

Снимок экрана 2021-10-28 в 8 45 04 PM

### CODE REVIEW

File: src/pages/candidates/Candidates.jsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import listOfCandidates from '../../mocks/listOfCandidates.json';
+import { tableFields } from '../../constants';
+import { getFieldLabel } from '../../utils';
+
+const Candidates = () => {
+  const [gridApi, setGridApi] = useState(null);
+  const onFirstDataRendered = (params) => {
+    params.api.sizeColumnsToFit();
+    window.setTimeout(() => {
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:15]: window.setTimeout ?? why ?

File: src/pages/candidates/Candidates.jsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    listOfCandidates.forEach((data, index) => {
+      const list = data;
+      list.id = index;
+    });
+    params.api.setRowData(listOfCandidates);
+  };
+
+  let sortActive = false;
+  let filterActive = false;
+
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:33]: Please use state

File: src/pages/candidates/Candidates.jsx (hunk 1 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx]: done

File: src/pages/candidates/Candidates.jsx (hunk 2 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx]: done

File: src/pages/candidates/Candidates.jsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+import { AgGridColumn, AgGridReact } from 'ag-grid-react';
+import 'ag-grid-enterprise';
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import listOfCandidates from '../../mocks/listOfCandidates.json';
+import { tableFields } from '../../constants';
+import { getFieldLabel } from '../../utils';
+
+const Candidates = () => {
+  const [gridApi, setGridApi] = useState(null);
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:12]:
```suggestion
  const [gridApi, setGridApi] = useState();
```

File: src/pages/candidates/Candidates.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+  const onFirstDataRendered = (params) => {
+    const colIds = params.columnApi.getAllColumns().map((c) => c.colId);
+    params.columnApi.autoSizeColumns(colIds);
+  };
+
+  const onGridReady = (params) => {
+    setGridApi(params.api);
+
+    listOfCandidates.forEach((data, index) => {
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:21]:
```suggestion
    listOfCandidates.forEach((listItem, index) => {
```

File: src/pages/candidates/Candidates.jsx [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+      const list = data;
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:22]: delete

File: src/pages/candidates/Candidates.jsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+import 'ag-grid-community/dist/styles/ag-grid.css';
+import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
+import listOfCandidates from '../../mocks/listOfCandidates.json';
+import { tableFields } from '../../constants';
+import { getFieldLabel } from '../../utils';
+
+const Candidates = () => {
+  const [gridApi, setGridApi] = useState(null);
+  const onFirstDataRendered = (params) => {
+    const colIds = params.columnApi.getAllColumns().map((c) => c.colId);
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:14]: Instead of colIds and c use full names

File: src/pages/candidates/Candidates.jsx [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+    gridApi.setSuppressRowDrag(sortActive);
+  };
+
+  const onFilterChanged = () => {
+    const filterActive = gridApi.isAnyFilterPresent();
+    gridApi.setSuppressRowDrag(filterActive);
+  };
+
+  const onRowDragMove = (event) => {
+    function moveInArray(arr, fromIndex, toIndex) {
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:40]: I don't like the name function. 

Also you can move 'moveInArray'  outside 'onRowDragMove'. Add return

File: src/pages/candidates/Candidates.jsx (hunk 3 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx:12]: done

File: src/pages/candidates/Candidates.jsx [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+  };
+
+  const paginationNumberFormatter = (params) =>
+    `[${params.value.toLocaleString()}]`;
+
+  return (
+    <Box>
+      <Box width="80px" marginBottom="5px" marginTop="30px">
+        <FormControl fullWidth variant="filled">
+          <InputLabel>Page Size</InputLabel>
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:74]: Use getFieldLabel()

File: src/pages/candidates/Candidates.jsx [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+          <Select defaultValue="10" label="10" onChange={onPageSizeChanged}>
+            <MenuItem value={10}>10</MenuItem>
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:76]: Please use array.

Just example:

image

File: src/pages/candidates/Candidates.jsx (hunk 6 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx:14]: done

File: src/pages/candidates/Candidates.jsx (hunk 8 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx]: done

File: src/pages/candidates/Candidates.jsx (hunk 9 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx]: done

File: src/pages/candidates/Candidates.jsx [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    </MenuItem>
+  ));
+
+  return (
+    <Box>
+      <Box width="80px" marginBottom="5px" marginTop="30px">
+        <FormControl fullWidth variant="filled">
+          <InputLabel>{getFieldLabel('candidates.form.inputLabel')}</InputLabel>
+          <Select defaultValue="10" label="10" onChange={onPageSizeChanged}>
+            {/* <MenuItem value={10}>10</MenuItem>
```
Comment (nsukhotskaya) [inline src/pages/candidates/Candidates.jsx:45]: Please delete unnecessary comment

**Review by itsnotpolza — DISMISSED** [review]

File: src/pages/candidates/Candidates.jsx (hunk 10 above)
Comment (ValeryiaMIRON) [reply src/pages/candidates/Candidates.jsx]: done

**Review by nsukhotskaya — APPROVED** [review]

**Review by oleksiuk-ol — APPROVED** [review]

**Pull request merged**